api/v1alpha1/api.go

```diff
@@ -122,6 +132,45 @@ type LLMRouteRuleMatch struct {
 	Headers []gwapiv1.HTTPHeaderMatch `json:"headers,omitempty"`
 }
 
+type LLMRouteFilterConfig struct {
```

Comment:
This indirection will help us deprecate and keep the API clean before/after extproc removal per #90 

fyi I will work on the compatibility policy doc next

Reply:
Also, this could be `AIGatewayFilterConfig` per #76

Reply:
right, let's change it all together.